### ISSUE TITLE
Fail to decompress these data streams ("lzma-alt" method?)

### ISSUE THREAD

**Issue by XVilka** [gendx/lzma-rs#25]
Unpacking the data, and noted that some streams can't be unpacked with lzma-rs, see the attached files of the compressed and uncompressed chunks.

[lzma_alt_samples.zip](https://github.com/gendx/lzma-rs/files/4096020/lzma_alt_samples.zip)

**Comment by gendx** [comment]
Could you document how standard this "Broadcom modification" is?

I won't have much time to investigate this in the following weeks, but feel free to send a pull-request to support this variant of the algorithm (preferably using a [decoding option](https://github.com/gendx/lzma-rs/blob/master/src/decode/options.rs) if there's something non-standard involved).

**Comment by XVilka** [comment]
I don't think it's a standard one, but quite common among compressed firmware images.

**Comment by voltagex** [comment]
@XVilka are you able to share how you decompressed these?

**Comment by XVilka** [comment]
@voltagex you can see the https://github.com/firmadyne/sasquatch/tree/master/LZMA/lzmalt

**Comment by voltagex** [comment]
@XVilka yes, this creates a library, not a program - did you write something to use this library to decompress the data?
